astropy/utils/data.py

```diff
@@ -2028,32 +2028,6 @@ def check_download_cache(pkgname="astropy"):
         raise CacheDamaged("\n".join(messages), bad_files=bad_files)
 
 
-@contextlib.contextmanager
-def _SafeTemporaryDirectory(suffix=None, prefix=None, dir=None):
-    """Temporary directory context manager.
-
-    This will not raise an exception if the temporary directory goes away
-    before it's supposed to be deleted. Specifically, what is deleted will
-    be the directory *name* produced; if no such directory exists, no
-    exception will be raised.
-
-    It would be safer to delete it only if it's really the same directory
-    - checked by file descriptor - and if it's still called the same thing.
-    But that opens a platform-specific can of worms.
-
-    It would also be more robust to use ExitStack and TemporaryDirectory,
-    which is more aggressive about removing readonly things.
-    """
-    d = mkdtemp(suffix=suffix, prefix=prefix, dir=dir)
-    try:
-        yield d
-    finally:
-        try:
-            shutil.rmtree(d)
-        except OSError:
```

Comment:
It's not as if there is any clever logic here; TemporaryDirectory certainly does at least as good a job, now that we can count on python 3.10.